File: LifeSamadhan/Backend/Models/ServiceProvider.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("serviceproviders")]
    public class ServiceProvider
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Column("Verified")]
        public bool Verified { get; set; } = false;

        [Required(ErrorMessage = "Availability is required")]
        [RegularExpression("^(AVAILABLE|BUSY|OFFLINE)$",
            ErrorMessage = "Availability must be AVAILABLE, BUSY or OFFLINE")]
        [Column("Availability")]
        public string Availability { get; set; } = "AVAILABLE";

        [Required(ErrorMessage = "UserId is required")]
        [Column("UserId")]
        public long UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User? User { get; set; }

        [Column("ProviderType")]
        public string? ProviderType { get; set; }

        [Column("HourlyRate")]
        public decimal? HourlyRate { get; set; }

        [Column("ServiceCategoryId")]
        public long? ServiceCategoryId { get; set; }

        [Column("City")]
        public string? City { get; set; }

        [Column("State")]
        public string? State { get; set; }

        [NotMapped]
        public int CompletedJobsCount { get; set; } = 0;

        [NotMapped]
        public int RejectedJobsCount { get; set; } = 0;

        [ForeignKey(nameof(ServiceCategoryId))]
        public ServiceCategory? ServiceCategory { get; set; }

        public List<ProviderLocation> Locations { get; set; } = new();

        public List<ProviderSkill> Skills { get; set; } = new();
    }
}

File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/ServiceProvider.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

import java.util.List;

@Entity
@Table(name = "service_providers")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ServiceProvider {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    @Column(nullable = false)
    private Boolean verified = false;

    @NotBlank(message = "Availability is required")
    @Pattern(regexp = "^(AVAILABLE|BUSY|OFFLINE)$", message = "Availability must be AVAILABLE, BUSY or OFFLINE")
    @Builder.Default
    @Column(nullable = false)
    private String availability = "AVAILABLE";

    @NotNull(message = "UserId is required")
    @Min(value = 1, message = "UserId must be a valid value")
    @Column(name = "user_id", nullable = false, unique = true)
    private Long userId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @OneToMany(mappedBy = "provider", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ProviderLocation> locations;

    @OneToMany(mappedBy = "provider", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<ProviderSkill> skills;

    @Builder.Default
    @Column(name = "completed_jobs_count", nullable = false)
    private Integer completedJobsCount = 0;

    @Builder.Default
    @Column(name = "rejected_jobs_count", nullable = false)
    private Integer rejectedJobsCount = 0;

    @Column(name = "hourly_rate")
    private Double hourlyRate;

    @Column(name = "provider_type")
    private String providerType;

    @Column(name = "service_category_id")
    private Long serviceCategoryId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "service_category_id", insertable = false, updatable = false)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private ServiceCategory category;
}